lib/catalog/owid/catalog/meta.py

```diff
@@ -56,6 +65,55 @@ class Source:
     def to_dict(self) -> Dict[str, Any]:
         ...
 
+    @staticmethod
+    def from_dict(d: Dict[str, Any]) -> "Source":
+        ...
+
+    def update(self, **kwargs: Dict[str, Any]) -> None:
+        for key, value in kwargs.items():
+            if value is not None:
+                setattr(self, key, value)
+
+
+@pruned_json
+@dataclass_json
+@dataclass
+class Origin:
+    # Original title from the source
+    dataset_title_owid: Optional[str] = None
+    # Our description of the dataset
+    dataset_description_owid: Optional[str] = None
+    # The description for this dataset used by the producer
+    dataset_description_producer: Optional[str] = None
+    # The name of the institution (without a year) or the main authors of the paper
+    producer: Optional[str] = None
+    # The full citation that the producer asks for
+    citation_producer: Optional[str] = None
+    # The authorative URL of the dataset
+    dataset_url_main: Optional[str] = None
+    # Direct URL to download the dataset
+    dataset_url_download: Optional[str] = None
+    # The URL of the dataset on our website
+    date_accessed: Optional[str] = None
+    # Either date or year of publication
```

Comment:
Maybe specify `# Publication date or, if the exact date is not known, publication year`.